Errors related to Float80 when  trying to compile on Raspberry Pi 3

I am trying to compile a vapor project that uses this repository on a Raspberry Pi 3 (w/ Ubuntu 16.04) but it fails with the errors below.
I might be wrong but from what I was able to find, Float80 is not available for ARM-based platforms. 

Is there a way to overcome this error?

Thanks in advance!

The errors I get are: 
/home/xxx/yyy-master/.build/checkouts/mysql.git-1505813759/Sources/MySQL/Connection/MySQLData.swift:219:24: error: use of unresolved identifier 'Float80'
            .flatMap { Float80($0) }
                       ^~~~~~~
Swift.Float:1:15: note: did you mean 'Float'?
public struct Float {
              ^
Swift.Float32:1:18: note: did you mean 'Float32'?
public typealias Float32 = Float
                 ^
Swift.Float64:1:18: note: did you mean 'Float64'?
public typealias Float64 = Double
                 ^
SwiftGlibc.float_t:1:18: note: did you mean 'float_t'?
public typealias float_t = Float
                 ^
/home/xxx/yyy-master/.build/checkouts/mysql.git-1505813759/Sources/MySQL/Connection/MySQLData.swift:237:36: error: use of unresolved identifier 'Float80'
                        .flatMap { Float80($0) }
                                   ^~~~~~~
Swift.Float:1:15: note: did you mean 'Float'?
public struct Float {
              ^
Swift.Float32:1:18: note: did you mean 'Float32'?
public typealias Float32 = Float
                 ^
Swift.Float64:1:18: note: did you mean 'Float64'?
public typealias Float64 = Double
                 ^
SwiftGlibc.float_t:1:18: note: did you mean 'float_t'?
public typealias float_t = Float

--- a/Sources/MySQL/Connection/MySQLData.swift
+++ b/Sources/MySQL/Connection/MySQLData.swift
@@ -215,9 +215,16 @@ public struct MySQLData: Equatable, Encodable {
     /// Access the value as an binary floating point.
     public func float<F>(_ type: F.Type) -> F? where F: BinaryFloatingPoint {
         switch storage {
-        case .text(let data): return data.flatMap { String(data: $0, encoding: .ascii) }
-            .flatMap { Float80($0) }
-            .flatMap { F.init($0) }
+        case .text(let data):
+            #if arch(arm) || arch(arm64)
+            return data.flatMap { String(data: $0, encoding: .ascii) }
+                .flatMap { Float($0)}
+                .flatMap { F.init($0) }
+            #else
+            return data.flatMap { String(data: $0, encoding: .ascii) }
+                .flatMap { Float80($0)}
+                .flatMap { F.init($0) }
+            #endif
         case .binary(let binary):
             switch binary.storage {
             case .integer1(let int8): return F(int8)
@@ -233,9 +240,15 @@ public struct MySQLData: Equatable, Encodable {
             case .string(let data):
                 switch binary.type {
                 case .MYSQL_TYPE_VARCHAR, .MYSQL_TYPE_VAR_STRING, .MYSQL_TYPE_STRING, .MYSQL_TYPE_DECIMAL, .MYSQL_TYPE_NEWDECIMAL:
+                    #if arch(arm) || arch(arm64)
+                    return String(data: data, encoding: .ascii)
+                        .flatMap { Float($0) }
+                        .flatMap { F.init($0) }
+                    #else
                     return String(data: data, encoding: .ascii)
                         .flatMap { Float80($0) }
                         .flatMap { F.init($0) }
+                    #endif
                 default: return nil // TODO: support more
                 }
             default: return nil

